### ISSUE TITLE
[OSX] link failed in plugin in_collectd 

### ISSUE THREAD

**Issue by jonnyyu** [fluent/fluent-bit#1561]
## Bug Report

**Describe the bug**

seems in_collectd doesn't build on mac.

**To Reproduce**
- Steps to reproduce the problem:

build with HEAD commit at that time: 8458c932
On Mac, install command line tools, install bison to 3.4 with HomeBrew and prepend bison path before system bison.
```
cd build
cmake .. 
```
```
Undefined symbols for architecture x86_64:
  "_be16toh", referenced from:
      _netprot_to_msgpack in libflb-plugin-in_collectd.a(netprot.c.o)
      _netprot_pack_value in libflb-plugin-in_collectd.a(netprot.c.o)
  "_be64toh", referenced from:
      _netprot_to_msgpack in libflb-plugin-in_collectd.a(netprot.c.o)
      _netprot_pack_value in libflb-plugin-in_collectd.a(netprot.c.o)
ld: symbol(s) not found for architecture x86_64
```

**Expected behavior**

**Screenshots**

**Your Environment**

* Version used:
* Configuration:
* Environment name and version (e.g. Kubernetes? What version?):
* Server type and version:
* Operating System and version:
* Filters and plugins:

**Additional context**

**Comment by edsiper** [comment]
cc: @fujimotos

**Comment by fujimotos** [comment]
I'm looking at this issue. It turned out that byteorder-family functions are
not really portable across UNIX platforms.

I'm going to fix this issue this afternoon.

**Comment by nigels-com** [comment]
Confirming that on my mac too.

Found a similar issue with a plausible fix: https://github.com/zeromq/zmqpp/issues/164

Also relevant for this: [libkern/OSByteOrder.h](https://opensource.apple.com/source/xnu/xnu-344.2/libkern/libkern/OSByteOrder.h.auto.html)

**Comment by edsiper** [comment]
thanks, fixed on #1564
